server/linter/plugins/plugins_entities.go

```diff
@@ -0,0 +1,62 @@
+package plugins
+
+import (
+	"context"
+
+	"github.com/kubeshop/tracetest/server/linter/analyzer"
+	"github.com/kubeshop/tracetest/server/linter/rules"
+	"github.com/kubeshop/tracetest/server/model"
+)
+
+type BasePlugin struct {
+	id           string
+	ruleRegistry rules.RuleRegistry
+}
+
+func NewPlugin(id string, ruleRegistry rules.RuleRegistry) Plugin {
+	return BasePlugin{id, ruleRegistry}
+}
+
+func (p BasePlugin) Id() string {
+	return p.id
+}
+
+func (p BasePlugin) RuleRegistry() rules.RuleRegistry {
+	return p.ruleRegistry
+}
+
+func (p BasePlugin) Execute(ctx context.Context, trace model.Trace, config analyzer.LinterPlugin) (analyzer.PluginResult, error) {
```

Comment:
All current plugins have the same execution logic, so instead of having to create a new `plugin` file for each, we use the default, and if needed, we can create specific plugin files implementing the `Plugin` interface to override execute method with custom logic